MiaServiceDotNetLibrary/Logging/RequestResponseLoggingMiddleware.cs

```diff
@@ -2,28 +2,46 @@
 using System.Diagnostics;
 using System.IO;
 using System.Threading.Tasks;
+using System.Collections.Generic;
 using MiaServiceDotNetLibrary.Logging.Entities;
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Http.Extensions;
 using log4net;
 
 namespace MiaServiceDotNetLibrary.Logging
 {
+    public class MiddlewareOptions
+{
+    public List<string> excludedPrefixes { get; set; } = new List<string>();
+}
     public class RequestResponseLoggingMiddleware
     {
         private readonly RequestDelegate _next;
+        private readonly MiddlewareOptions _options;
+
         internal static readonly string  RequestIdDictionaryKey = "requestId";
         internal static readonly string  RequestIdHeaderKey = "x-request-id";
         internal static readonly string  UserAgentHeaderKey = "User-Agent";
         internal static readonly string  ForwardedHostHeaderKey = "x-forwarded-host";
 
-        public RequestResponseLoggingMiddleware(RequestDelegate next)
+        public RequestResponseLoggingMiddleware(RequestDelegate next, MiddlewareOptions options)
         {
             _next = next;
+            _options = options;
         }
         
         public async Task Invoke(HttpContext context)
         {
+            string path = context.Request.Path;
+            foreach (string prefix in _options.excludedPrefixes)
```

Comment:
define a function that does this checks so that the code will look like:

```
if (shouldExcludePrefix( ... )) {
       await _next(context);
       return;    
}
```

in alternative, there is no API to search in a list with a custom search function?

Reply:
Done! I have simplified using Exists() List method!